lightning/src/ln/channelmanager.rs

```diff
@@ -119,6 +121,12 @@ pub(super) struct HTLCPreviousHopData {
 	incoming_packet_shared_secret: [u8; 32],
 }
 
+struct ClaimableHTLC {
+	src: HTLCPreviousHopData,
+	value: u64,
+	payment_data: Option<msgs::FinalOnionHopData>,
```

Comment:
Wondering if we could make this non-optional? Would be nice because then we could get rid of the semi-redundant `value` field

Reply:
Hmmmm, first of all `value` != `payment_data.total_msat` - `value` is the value of this HTLC, `total_msat` is the total value of the payment (which is different for MPP). We could refactor the total_msat out into a second field here, but payment_secret, ultimately, is gonna have to remain an Option<>. It seems a bit simpler to me to use the existing msgs field, but I'm open to looking harder.

Reply:
Makes sense. Could you add a comment saying where/when this struct is supposed to be used?